contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,910 @@
+        emit PrincipalPositionBeforeSwap(
+            _wethPositionBalance,
+            _oethbPositionBalance
+        );
+
+        // Swap it
+        swapRouter.exactInputSingle(
+            // sqrtPriceLimitX96 is just a rough sanity check that we are within 0 -> 1 tick
+            // a more fine check is performed in _checkForExpectedPoolPrice
+            // TBD(!): this needs further work if we want to generalize this approach
+            ISwapRouter.ExactInputSingleParams({
+                tokenIn: address(_tokenToSwap),
+                tokenOut: _swapWeth ? OETHb : WETH,
+                tickSpacing: tickSpacing, // set to 1
+                recipient: address(this),
+                deadline: block.timestamp,
+                amountIn: _amountToSwap,
+                amountOutMinimum: _minTokenReceived, // slippage check
+                sqrtPriceLimitX96: _swapWeth
+                    ? sqrtRatioX96Tick0
+                    : sqrtRatioX96Tick1
+            })
+        );
+
+        // emit an event so it is easier to find correct values off-chain
+        (_wethPositionBalance, _oethbPositionBalance) = getPositionPrincipal();
+        emit PrincipalPositionAfterSwap(
+            _wethPositionBalance,
+            _oethbPositionBalance
+        );
+    }
+
+    /**
+     * @dev Add liquidity into the pool in the pre-configured WETH to OETHb share ratios
+     * configured by the `poolWethShare` property. This function will respect liquidity
+     * ratios when there no liquidity yet in the pool. If liquidity is already present
+     * then it relies on the `_swapToDesiredPosition` function in a step before to already
+     * move the trading price to desired position (with some tolerance).
+     */
+    // rebalance already has re-entrency checks
+    // slither-disable-start reentrancy-no-eth
+    function _addLiquidity() internal {
+        uint256 _wethBalance = IERC20(WETH).balanceOf(address(this));
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        require(_wethBalance > 0, "Must add some WETH");
+
+        uint160 _currentPrice = getPoolX96Price();
+        // sanity check active trading price is positioned within our desired tick
+        require(
+            _currentPrice > sqrtRatioX96Tick0 &&
+                _currentPrice < sqrtRatioX96Tick1,
+            "Not in expected tick range"
+        );
+
+        // in case oethb would be the 1st token we'd need to call estimateAmount0 here
+        uint256 _oethbRequired = helper.estimateAmount1(
+            _wethBalance,
+            address(clPool),
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        if (_oethbRequired > _oethbBalance) {
+            IVault(vaultAddress).mintForStrategy(
+                _oethbRequired - _oethbBalance
+            );
+        }
+
+        uint256 _wethAmountSupplied;
+        uint256 _oethbAmountSupplied;
+        if (tokenId == 0) {
+            (
```

Comment:
Instead of doing this:

```
(
                uint256 _mintedTokenId,
                ,
                uint256 __wethAmountSupplied,
                uint256 __oethbAmountSupplied
            )
```
then this
```
            _wethAmountSupplied = __wethAmountSupplied;
            _oethbAmountSupplied = __oethbAmountSupplied;
            tokenId = _mintedTokenId;
```

Could we just do? 

```
(tokenId,,_wethAmountSupplied, _oethbAmountSupplied)
```
This removes the need for the temp variables that then mostly set other temp variables. 

Same for the following `increaseLiquidity`

If you passionately prefer the currently style, we can keep it.

Reply:
Yeah I had some issues here getting things to compile. Figured out what the problem is. When deconstructuring multiple variable returns you either need to declare all the variables or none of them. Only assigning some and declaring others doesn't compile. So this won't compile: 
<img width="269" alt="Screenshot 2024-08-23 at 11 47 06" src="https://github.com/user-attachments/assets/cc54dcb6-234d-4b96-97ea-5840dca07d96">

while this will: 
<img width="258" alt="Screenshot 2024-08-23 at 11 47 24" src="https://github.com/user-attachments/assets/b609ca53-4a9c-4687-9c97-8ed16786a778">

Simplifying the code here: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/02ae07de8a957ea5ffd0c42e3c763fe2101bfa07

Reply:
Yeah, it's just one of these weird solidity things that don't actually make sense.